lib/rules/forbid-component-props.js

```diff
@@ -78,7 +78,12 @@ module.exports = {
 
     return {
       JSXAttribute(node) {
-        const tag = node.parent.name.name;
+        const parentName = node.parent.name;
+        let tag = parentName.name;
+        if (!tag && parentName.object.name !== 'this') {
```

Comment:
why not `this`? iow, `<this.Foo />` should be valid as well, no?

Reply:
One test was failing, so I thought there is some reason I am not aware of behind forbidding it.

Reply:
Why would it be a different component every time? `this.Component` could just be an own property on the instance, the same every time.

I think it's reasonable to assume that `this` inside a class method is always the class instance.

Reply:
I've taken care of it; it's that the "first letter is capitalized" check wasn't yet taking into account the namespace and the dot.